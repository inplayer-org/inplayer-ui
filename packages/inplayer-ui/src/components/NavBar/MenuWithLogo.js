// @flow
import React, { type Node, type Element } from 'packages/inplayer-ui/src/components/NavBar/react';
import { LogoMenuContainer, NavbarLogo, MenuContainer } from './styled';

type Props = {
  logo: ?Element<*> | ?string,
  children: Node,
};

const MenuWithLogo = ({ logo, children }: Props) => (
  <LogoMenuContainer>
    <NavbarLogo src={logo} />
    <MenuContainer>{children}</MenuContainer>
  </LogoMenuContainer>
);

export default MenuWithLogo;
