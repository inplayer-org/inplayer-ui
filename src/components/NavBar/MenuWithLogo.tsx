import React, { ReactNode } from 'react';
import { LogoMenuContainer, NavbarLogo, MenuContainer } from './styled';

type Props = {
  logo?: string;
  children: ReactNode;
};

const MenuWithLogo = ({ logo, children }: Props) => (
  <LogoMenuContainer>
    <NavbarLogo src={logo} />
    <MenuContainer>{children}</MenuContainer>
  </LogoMenuContainer>
);

export default MenuWithLogo;
