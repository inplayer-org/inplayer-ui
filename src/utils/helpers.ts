import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
