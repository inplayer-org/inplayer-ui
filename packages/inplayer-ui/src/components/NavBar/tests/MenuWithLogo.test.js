import { generateSnapshots } from 'packages/inplayer-ui/src/components/NavBar/tests/helpers';
import MenuWithLogo from '../MenuWithLogo';

generateSnapshots(MenuWithLogo);
