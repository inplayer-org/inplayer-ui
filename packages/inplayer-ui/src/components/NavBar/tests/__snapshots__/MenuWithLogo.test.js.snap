// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuWithLogo renders correctly 1`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 34px;
  margin: 0.5rem 0;
}

.c2 {
  width: 100%;
  text-align: left;
  margin: 3.125rem 0 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<span
  className="c0"
>
  <img
    className="c1"
  />
  <nav
    className="c2"
  />
</span>
`;
